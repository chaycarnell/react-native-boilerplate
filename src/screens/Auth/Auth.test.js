import React from 'react';
import { fireEvent, render, cleanup } from '@testing-library/react-native';
import AuthScreen from './Auth';

afterEach(cleanup);

describe('Auth screen', () => {
  describe('given default props', () => {
    let renderedComponent;

    const testProps = {
      navigation: {
        navigate: jest.fn(),
      },
    };

    beforeEach(() => {
      renderedComponent = render(
        <AuthScreen navigation={testProps.navigation} />,
      );
    });

    it('should render correctly', () => {
      const { toJSON } = renderedComponent;
      expect(toJSON()).toMatchSnapshot();
    });

    it('should navigate to dashboard page when touchable highlight is selected', () => {
      const { getByTestId } = renderedComponent;
      fireEvent.press(getByTestId('AuthNavigation'));

      expect(testProps.navigation.navigate).toHaveBeenCalled();
      expect(testProps.navigation.navigate).toHaveBeenCalledWith(
        'Dashboard',
        {},
      );
    });
  });
});
