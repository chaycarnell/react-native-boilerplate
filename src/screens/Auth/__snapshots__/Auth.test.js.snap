// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth screen given default props should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="AuthNavigation"
  >
    <View
      color="#0f62fe"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#0f62fe",
            "display": "flex",
            "height": 667,
            "justifyContent": "center",
            "opacity": 1,
            "width": 750,
          },
        ]
      }
    >
      <Text
        color="#FFF"
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 48,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
